pkg/api/authz.go

```diff
@@ -213,6 +208,12 @@ func AuthzHandler(ctlr *Controller) mux.MiddlewareFunc {
 				return
 			}
 
+			if strings.Contains(request.RequestURI, constants.ExtSearchPrefix) {
+				next.ServeHTTP(response, request.WithContext(ctx))
```

Comment:
Why are we not falling back to the normal path below?

Reply:
And what is the "next" handler?

Reply:
We are not falling back on the normal path because on line [239](https://github.com/project-zot/zot/pull/728/files/7517df3258d350ac53c0d84be6d91f02f6575aed#diff-fa881d434bc95a6195440f2122784dacca2bcbc19b93baf9e575feed1265340aL239) the "resource" variable will be empty, since this in a request variable for REST API calls. Thus authz will fail - line 241.

Reply:
> And what is the "next" handler?

idk what do you mean by this, nothing has changed in this approach. 
```func(next http.Handler) http.Handler``` is the signature of a MiddlewareFunc, and in our case the "next" handler will be the router inside the Controller.
We serve back the incoming http requests with modified context ```next.ServeHTTP(response, request.WithContext(ctx))```